jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java

```diff
@@ -125,13 +125,7 @@ public UnwrittenLayer build() throws IOException {
     return new UnwrittenLayer(tarStreamBuilder.toBlob());
   }
 
-  public List<Path> getSourceFiles() {
-    List<Path> allSourceFiles = new ArrayList<>();
-
-    for (LayerEntry layerEntry : layerEntries) {
-      allSourceFiles.addAll(layerEntry.getSourceFiles());
-    }
-
-    return allSourceFiles;
+  public ImmutableList<LayerEntry> getLayerEntries() {
```

Comment:
This isn't really related to this PR, but I'm a little behind.

I wonder if `ReprocibleLayerBuilder` should not have `build()` and `getLayerEntries()` called on it, perhaps `CacheWriter#writeLayer` should take a `ReproducibleLayer` type (from `ReproducibleLayer.builder()`) and have `unwrittenLayer` and `layerEntries` methods on it. Only because we can't programmatically guarantee that there are no changes to ReproducibleLayerBuilder between calls to `build` and `getLayerEntries`.

Reply:
Agreed. Another way is to create an interface that's like `LayerBuilder` and `CacheWriter#writeLayer` will take a `LayerBuilder` (with `ReproducibleLayerBuilder` being an interface). All implementations of `LayerBuilder` should be immutable them being built with some builder nested class.